packages/cli/init/src/utils/docsUrlUtils.ts

```diff
@@ -0,0 +1,70 @@
+import axios from "axios";
+import { load } from "cheerio";
+
+export type DocsUrl = ReadMeDocsUrl;
+
+export interface DocsUrlInterface {
+    getAllOpenApiUrls(): Promise<string[]>;
+    getBaseUrl(): string;
+    isUrlValid(): boolean;
+}
+
+class DocsUrlBase implements DocsUrlInterface {
+    url: string;
+
+    constructor(url: string) {
+        this.url = url;
+    }
+
+    async getAllOpenApiUrls(): Promise<string[]> {
+        return [];
+    }
+
+    getBaseUrl(): string {
+        const url = new URL(this.url);
+        return url.origin;
+    }
+
+    isUrlValid(): boolean {
+        if (!this.url || this.url === "") {
+            return false;
+        }
+
+        try {
+            new URL(this.url);
+            return true;
+        } catch (err) {
+            return false;
+        }
+    }
+}
+
+export class ReadMeDocsUrl extends DocsUrlBase {
+    type: "readme" = "readme";
+
+    async getAllOpenApiUrls(): Promise<string[]> {
+        const origin = this.getBaseUrl();
+        const response = await axios.get(origin + "/openapi");
```

Comment:
i would make sure you are safely joining and the leading `/` does not conflict

Reply:
ah yes, accomplished this with a URL builder!